packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/launchpad/curve.ts

```diff
@@ -0,0 +1,322 @@
+import {
+  BaseFeeMode,
+  CollectFeeMode,
+  MigrationOption,
+  MigrationFeeOption,
+  TokenType,
+  CreateConfigParam,
+  ActivationType
+} from '@meteora-ag/dynamic-bonding-curve-sdk'
+import { Keypair, PublicKey } from '@solana/web3.js'
+import BN from 'bn.js'
+
+/**
+ * Makes a production curve for dbc deployment
+ * @param params Object containing:
+ *   - payer: the keypair of the payer
+ *   - configKey: the keypair for the config
+ *   - partner: the public key of the partner
+ *   - rewardPool: the public key of the reward pool
+ * @returns the curve design
+ */
+export const makeCurve = ({
+  payer,
+  configKey,
+  partner,
+  rewardPoolTokenAccount
+}: {
+  payer: Keypair
+  configKey: Keypair
+  partner: PublicKey
+  rewardPoolTokenAccount: PublicKey
+}): CreateConfigParam => ({
+  payer: payer.publicKey,
+  config: configKey.publicKey,
+  feeClaimer: partner,
+  leftoverReceiver: rewardPoolTokenAccount,
+  quoteMint: new PublicKey('9LzCMqDgTKYz9Drzqnpgee3SGa89up3a247ypMj2xrqM'),
+
+  // Fees
+  poolFees: {
+    baseFee: {
+      cliffFeeNumerator: new BN(10000000),
+      baseFeeMode: BaseFeeMode.FeeSchedulerLinear,
+      firstFactor: 0,
+      secondFactor: new BN(0),
+      thirdFactor: new BN(0)
+    },
+    dynamicFee: null
+  },
+  activationType: ActivationType.Slot,
+  collectFeeMode: CollectFeeMode.QuoteToken,
+
+  // Graduation
+  migrationOption: MigrationOption.MET_DAMM_V2,
+  migrationQuoteThreshold: new BN('20000000000000'),
+  migrationFeeOption: MigrationFeeOption.FixedBps100,
+  // Migrated pool fee is unused because fee option is set above
+  migratedPoolFee: {
+    collectFeeMode: CollectFeeMode.QuoteToken,
+    dynamicFee: 0,
+    poolFeeBps: 0
+  },
+  migrationFee: {
+    feePercentage: 0,
+    creatorFeePercentage: 0
+  },
+
+  // Token params
+  tokenType: TokenType.SPL,
+  tokenDecimal: 9,
+
+  // LP design
+  partnerLpPercentage: 0,
+  creatorLpPercentage: 0,
+  partnerLockedLpPercentage: 50,
+  creatorLockedLpPercentage: 50,
+  creatorTradingFeePercentage: 50,
+
+  // Curve design
+  // Calculation worksheet: https://colab.research.google.com/drive/1hZy7CEekEF3LGi81IWHm_1DL9Pjd6t6U
```

Comment:
not sure if intentional but i dont have access to view this

Reply:
updated link